src/lib.rs

```diff
@@ -381,7 +390,12 @@ struct FileRuntime {
 
 impl FileRuntime {
     fn new(expect: &Expect) -> FileRuntime {
-        let path = WORKSPACE_ROOT.join(expect.position.file);
+        let path = match WORKSPACE_ROOT.as_deref() {
+            Ok(root) => root.join(expect.position.file),
```

Comment:
Hm, should we have similar is_absolute logic here?

Reply:
I'm not sure, aren't those potentially source files that need updating. No idea how to get their path without WORKSPACE_ROOT. Or could those path be absolute too?

Reply:
Hm, I still don’t understand why we need two different algorithms to get an absolute file path. Why in one case we do “if absolute, use it, otherwise join to ws path (and panic if there’s none)” and in the other we do “panic if there is no ws path and join to it”

Reply:
Maybe I'm misunderstanding the logic here. My understanding is, the abs_path place is used when the expect_tests are executed, which is either provided a string or a path to a file. Here we can assume its reasonably feasible for the user to only provide absolute paths. In contrast this other place in FileRuntime is called when the user specifies UPDATE_EXPECT, in which case it needs to find paths to the expect files *and* the Rust source files. The  Rust source files are required as UPDATE_EXPECT will also update string literals. How would the user provide absolute paths for the Rust source files? Isn't WORKSPACE_ROOT used to find the paths to those Rust files?

Reply:
The path is provided via `file!()` macro, which presumably *could* return absolute paths.

Reply:
To clarify, I think practically this *doesnt* matter, as today those paths are always relative. I am worrying here about code-quality aspects: we do the same operation, joining a path to ws root, twice, and we use slightly different logic in two cases. Unless there’s a strong reason why the logic *must* be different, I would prefer this to be coded just once.

that’s why I think we maybe don’t need a global static, and need just 

```rust
fn to_abs_path(p: &Path) {
   static WS_ROOT: …
}
```

Reply:
Ok will update.